### PR TITLE
Refactor CLV build_model logic

### PR DESCRIPTION
**Pull request by ricardoV94**
## Description

* Move distribution initialization away from `CLVModels.__init__`
* Add helper to validate columns
* Get rid of many properties in favor of self.data

## Related Issue

- [x] Related to #527 
- [x] Related to #409  
- [x] Related to #318

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [x] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [ ] MMM
- [x] CLV

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [x] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--564.org.readthedocs.build/en/564/

### CODE REVIEW

**Comment by ColtAllen** [comment]
There is a `build_model` for every child model class - should an abstract method be added to the base `CLVModel` class to enforce that convention?

File: pymc_marketing/clv/models/pareto_nbd.py [hunk 1]
```diff
@@ -166,44 +166,21 @@ def __init__(
...
-            raise KeyError("recency column is missing from data")
-        try:
-            self.T = data["T"]
-        except KeyError:
-            raise KeyError("T column is missing from data")
-
+        warnings.warn(
+            "The Pareto/NBD model is still experimental. Please see code examples in documentation if model fitting issues are encountered.",
+            UserWarning,
+        )
```
Comment (ColtAllen) [inline pymc_marketing/clv/models/pareto_nbd.py:172]: Why is this warning being re-added? For https://github.com/pymc-labs/pymc-marketing/pull/545?

Comment (ricardoV94) [reply pymc_marketing/clv/models/pareto_nbd.py:172]: Probably a git conflict mistake on my part. I started working on this PR on a super stale main branch xD. Will revert

**Comment by ColtAllen** [comment]
I like how this cleans up `__init__` in all the models quite a bit. The new `_validate_cols` method is also relevant to https://github.com/pymc-labs/pymc-marketing/issues/175.

File: pymc_marketing/clv/models/pareto_nbd.py (hunk 1 above)
Comment (ricardoV94) [reply pymc_marketing/clv/models/pareto_nbd.py]: Fixed

**Review by ColtAllen — APPROVED** [review]
All tests are passing and I couldn't find any issues when going through the code. I think this one is good to go!

**Pull request merged**